src/adapters/surrealDB/pipeline/postprocess/query/cleanQueryRes.ts

```diff
@@ -0,0 +1,95 @@
+import type { PipelineOperation, BQLResponseSingle, BormConfig } from '../../../../../types';
+import { produce } from 'immer';
+import type { SurrealDbResponse, EnrichedBqlQuery } from '../../../types/base';
+
+const processNull = (config: BormConfig, obj: Record<string, unknown>) => {
+	for (const key in obj) {
+		const value = obj[key];
+
+		if (config.query?.returnNulls) {
+			if (Array.isArray(value) && value.length === 0) {
+				obj[key] = null;
+				continue;
+			}
+		} else {
+			if (value === null) {
+				delete obj[key];
+				continue;
+			}
+
+			if (Array.isArray(value) && value.length === 0) {
+				delete obj[key];
+				continue;
+			}
+		}
+	}
+};
+
+const cleanUpObj = ({
+	config,
+	item,
+	query,
+}: {
+	query: EnrichedBqlQuery;
+	config: BormConfig;
+	item: BQLResponseSingle;
+}) => {
+	processNull(config, item);
+
+	// INTERNAL SYMBOLS
+	Object.getOwnPropertySymbols(item).forEach((symbol) => {
```

Comment:
I think this was already in the typedb query but i think we could use structuredClone or {...spread} instead, i think one of those don't copy the symbols and is an immutable way to remove those symbols

Reply:
I have just tried with both structuredClone and ...spread. For structuredClone, I notice that it will change the order of items in the array. For example,

```
{
  spaces: ['space-1', 'space-2', 'space-3']
}
```

would become the following if I used `structuredClone`

```
{
  spaces: ['space-2', 'space-1', 'space-3']
}
```

Obviously we can recursively sort all arrays after this clone, but at the moment the current solution seems to be doing the job with less code?

For spreading, it only handles 1 level, and it is an immutable way to code, which don't work as expected inside the `produce` callback, as immer expect you mutate the object.

So I think two options we can choose is 1. use `immer` and `produce`, and mutate the object or 2. deep clone the object and the transform it immutablely. What do you think?